agent/api/container/port_binding.go

```diff
@@ -30,9 +32,13 @@ const (
 // PortBinding represents a port binding for a container
 type PortBinding struct {
 	// ContainerPort is the port inside the container
-	ContainerPort uint16
+	ContainerPort *uint16
+	// ContainerPortRange is a range of ports exposed inside the container
+	ContainerPortRange *string
 	// HostPort is the port exposed on the host
 	HostPort uint16
+	// HostPortRange is the range of host ports corresponding to ContainerPortRange
+	HostPortRange string
```

Comment:
`containerPort` and `containerPortRange` are references while `hostPort` is not. This is because we want to perform nil checks on `containerPort` and `containerPortRange` while supplying these to docker and while generating the list of `networkBinding`.

The assumption here is that per port binding config, either `containerPort` or `containerPortRange` will be set, not both. This is validated on the ECS task definition level.